In [1]:
import pandas as pd
import numpy as np
import os

In [2]:
os.getcwd()

'C:\\Users\\TEST-USMT'

In [3]:
os.chdir("C:\\Users\\TEST-USMT\\Desktop\\personal.L")

In [4]:
os.getcwd()

'C:\\Users\\TEST-USMT\\Desktop\\personal.L'

In [6]:
data = pd.read_excel("FoodBalanceSheets_E_Africa_NOFLAG.xlsx")

In [7]:
data.head(10)

Area Code     Area  Item Code              Item  Element Code  \
0          4  Algeria       2501        Population           511   
1          4  Algeria       2501        Population          5301   
2          4  Algeria       2901       Grand Total           664   
3          4  Algeria       2901       Grand Total           674   
4          4  Algeria       2901       Grand Total           684   
5          4  Algeria       2903  Vegetal Products           664   
6          4  Algeria       2903  Vegetal Products           674   
7          4  Algeria       2903  Vegetal Products           684   
8          4  Algeria       2941   Animal Products           664   
9          4  Algeria       2941   Animal Products           674   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   
5           Food supply (kcal/capita/day)  kcal/capita/day   2932.00   
6  Protein supply quantity (g/capita/day)     g/capita/day     67.14   
7      Fat supply quantity (g/capita/day)     g/capita/day     53.84   
8           Food supply (kcal/capita/day)  kcal/capita/day    444.00   
9  Protein supply quantity (g/capita/day)     g/capita/day     27.76   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28  
5   2958.00   2941.00   2921.00   2932.00  
6     67.38     67.37     66.11     67.10  
7     54.72     52.04     55.59     54.57  
8    421.00    431.00    421.00    390.00  
9     26.97     27.35     26.71     24.73

In [9]:
data.shape

(60943, 12)

In [10]:
data.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [15]:
data.columns

Index(['Area Code', 'Area', 'Item Code', 'Item', 'Element Code', 'Element',
       'Unit', 'Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'],
      dtype='object')

In [19]:
data.tail(20)

Area Code      Area  Item Code           Item  Element Code  \
60923        181  Zimbabwe       2680    Infant food          5611   
60924        181  Zimbabwe       2680    Infant food          5072   
60925        181  Zimbabwe       2680    Infant food          5911   
60926        181  Zimbabwe       2680    Infant food          5301   
60927        181  Zimbabwe       2680    Infant food          5170   
60928        181  Zimbabwe       2680    Infant food          5142   
60929        181  Zimbabwe       2680    Infant food           645   
60930        181  Zimbabwe       2680    Infant food           664   
60931        181  Zimbabwe       2680    Infant food           674   
60932        181  Zimbabwe       2680    Infant food           684   
60933        181  Zimbabwe       2899  Miscellaneous          5611   
60934        181  Zimbabwe       2899  Miscellaneous          5072   
60935        181  Zimbabwe       2899  Miscellaneous          5911   
60936        181  Zimbabwe       2899  Miscellaneous          5301   
60937        181  Zimbabwe       2899  Miscellaneous          5170   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit  Y2014  Y2015  \
60923                         Import Quantity      1000 tonnes   0.00   0.00   
60924                         Stock Variation      1000 tonnes   0.00   0.00   
60925                         Export Quantity      1000 tonnes   0.00   0.00   
60926                Domestic supply quantity      1000 tonnes   0.00   0.00   
60927                               Residuals      1000 tonnes   0.00   0.00   
60928                                    Food      1000 tonnes   0.00   0.00   
60929     Food supply quantity (kg/capita/yr)               kg   0.00   0.00   
60930           Food supply (kcal/capita/day)  kcal/capita/day   0.00   0.00   
60931  Protein supply quantity (g/capita/day)     g/capita/day   0.00   0.00   
60932      Fat supply quantity (g/capita/day)     g/capita/day   0.00   0.00   
60933                         Import Quantity      1000 tonnes  42.00  47.00   
60934                         Stock Variation      1000 tonnes   0.00   0.00   
60935                         Export Quantity      1000 tonnes   1.00   1.00   
60936                Domestic supply quantity      1000 tonnes  42.00  46.00   
60937                               Residuals      1000 tonnes   0.00   0.00   
60938                                    Food      1000 tonnes  42.00  46.00   
60939     Food supply quantity (kg/capita/yr)               kg   3.06   3.33   
60940           Food supply (kcal/capita/day)  kcal/capita/day   3.00   4.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day   0.10   0.11   
60942      Fat supply quantity (g/capita/day)     g/capita/day   0.04   0.05   

       Y2016  Y2017  Y2018  
60923   0.00   0.00   0.00  
60924   0.00   0.00   0.00  
60925   0.00   0.00   0.00  
60926   0.00   0.00   0.00  
60927   0.00   0.00   0.00  
60928   0.00   0.00   0.00  
60929   0.00   0.00   0.01  
60930   0.00   0.00   0.00  
60931   0.00   0.00   0.00  
60932   0.00   0.00   0.00  
60933  33.00  15.00  16.00  
60934   0.00  -4.00   0.00  
60935   0.00   0.00   1.00  
60936  33.00  19.00  16.00  
60937   0.00   0.00   0.00  
60938  33.00  19.00  16.00  
60939   2.35   1.33   1.08  
60940   3.00   1.00   1.00  
60941   0.08   0.04   0.04  
60942   0.03   0.02   0.01

In [20]:
print(data[['Item', 'Element']])

                Item                                 Element
0         Population           Total Population - Both sexes
1         Population                Domestic supply quantity
2        Grand Total           Food supply (kcal/capita/day)
3        Grand Total  Protein supply quantity (g/capita/day)
4        Grand Total      Fat supply quantity (g/capita/day)
...              ...                                     ...
60938  Miscellaneous                                    Food
60939  Miscellaneous     Food supply quantity (kg/capita/yr)
60940  Miscellaneous           Food supply (kcal/capita/day)
60941  Miscellaneous  Protein supply quantity (g/capita/day)
60942  Miscellaneous      Fat supply quantity (g/capita/day)

[60943 rows x 2 columns]


# Question 11. Total sum of Animal Fat produced in 2014 and 2017 respectively

In [36]:
AnimalFat = data.groupby('Item').agg(['sum','count'])

In [38]:
AnimalFat.head(5)

Area Code        \
                          sum count   
Item                                  
Alcohol, Non-Food       38447   283   
Alcoholic Beverages     75016   564   
Animal Products         18060   135   
Animal fats             85279   639   
Apples and products     69280   517   

                                                                  Area        \
                                                                   sum count   
Item                                                                           
Alcohol, Non-Food    AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAngo...   283   
Alcoholic Beverages  AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   564   
Animal Products      AlgeriaAlgeriaAlgeriaAngolaAngolaAngolaBeninBe...   135   
Animal fats          AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   639   
Apples and products  AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   517   

                    Item Code       Element Code        \
                          sum count          sum count   
Item                                                     
Alcohol, Non-Food      752497   283      1515090   283   
Alcoholic Beverages   1649136   564      2171381   564   
Animal Products        397035   135        90990   135   
Animal fats           1882494   639      2567303   639   
Apples and products   1352989   517      1920418   517   

                                                               Element        \
                                                                   sum count   
Item                                                                           
Alcohol, Non-Food    ProductionImport QuantityStock VariationDomest...   283   
Alcoholic Beverages  ProductionImport QuantityStock VariationExport...   564   
Animal Products      Food supply (kcal/capita/day)Protein supply qu...   135   
Animal fats          ProductionImport QuantityStock VariationExport...   639   
Apples and products  ProductionImport QuantityStock VariationExport...   517   

                     ...      Y2014            Y2015            Y2016        \
                     ...        sum count        sum count        sum count   
Item                 ...                                                      
Alcohol, Non-Food    ...    2403.00   275    2180.00   271    2223.00   274   
Alcoholic Beverages  ...  102410.11   562   98783.72   559   96958.75   559   
Animal Products      ...   11935.65   135   11811.73   135   11661.69   135   
Animal fats          ...  209460.54   635  200675.72   633  183314.22   633   
Apples and products  ...    9499.23   510   10559.15   511    9853.26   512   

                         Y2017            Y2018        
                           sum count        sum count  
Item                                                   
Alcohol, Non-Food      2348.00   273    2293.00   273  
Alcoholic Beverages   95581.06   560   97847.27   561  
Animal Products       11547.65   135   11578.61   135  
Animal fats          269617.53   636  269648.27   636  
Apples and products   10198.90   511    9640.51   508  

[5 rows x 22 columns]

# Answer = Animal Fat for 2014 = 209460.54 and 2017 = 269617.53

# Question 12. Mean and standard deviation for 2015

In [41]:
data.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

# Answer = Mean(2015) = 135.235 and std(2015) = 1603.403

# Question 13. Total number and percentage of missing data in 2016 to 2 decimal places

In [43]:
percent_missing = data.isnull().sum() * 100 / len(data)

In [45]:
percent_missing

Area Code       0.000000
Area            0.000000
Item Code       0.000000
Item            0.000000
Element Code    0.000000
Element         0.000000
Unit            0.000000
Y2014           2.607354
Y2015           2.540078
Y2016           2.518747
Y2017           2.471162
Y2018           2.356300
dtype: float64

In [47]:
missing = data.isnull().sum()

In [48]:
missing

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

# Answer. Missing_Data(2016) = 1535, percent_missing = 2.52

# Question 14. which year had the highest correlation with Element code

In [52]:
data.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

# Answer. Highest Correlation by their element code is 0.024457 (2014)

# Question 15. Year with the highest form of import quantity

In [58]:
Import_quantity = data.groupby('Element').agg(['sum'])

In [59]:
Import_quantity.head(8)

Area Code  \
                                          sum   
Element                                         
Domestic supply quantity               708993   
Export Quantity                        599910   
Fat supply quantity (g/capita/day)     675050   
Feed                                   176272   
Food                                   663295   
Food supply (kcal/capita/day)          674057   
Food supply quantity (kg/capita/yr)    658446   
Import Quantity                        688174   

                                                                                  Area  \
                                                                                   sum   
Element                                                                                  
Domestic supply quantity             AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Export Quantity                      AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Fat supply quantity (g/capita/day)   AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Feed                                 AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Food                                 AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Food supply (kcal/capita/day)        AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Food supply quantity (kg/capita/yr)  AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Import Quantity                      AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   

                                    Item Code  \
                                          sum   
Element                                         
Domestic supply quantity             14197445   
Export Quantity                      11840553   
Fat supply quantity (g/capita/day)   13535000   
Feed                                  3538507   
Food                                 13285035   
Food supply (kcal/capita/day)        13511060   
Food supply quantity (kg/capita/yr)  13185401   
Import Quantity                      13795966   

                                                                                  Item  \
                                                                                   sum   
Element                                                                                  
Domestic supply quantity             PopulationCereals - Excluding BeerWheat and pr...   
Export Quantity                      Cereals - Excluding BeerWheat and productsRice...   
Fat supply quantity (g/capita/day)   Grand TotalVegetal ProductsAnimal ProductsCere...   
Feed                                 Cereals - Excluding BeerWheat and productsRice...   
Food                                 Cereals - Excluding BeerWheat and productsRice...   
Food supply (kcal/capita/day)        Grand TotalVegetal ProductsAnimal ProductsCere...   
Food supply quantity (kg/capita/yr)  Cereals - Excluding BeerWheat and productsRice...   
Import Quantity                      Cereals - Excluding BeerWheat and productsRice...   

                                    Element Code  \
                                             sum   
Element                                            
Domestic supply quantity                28068795   
Export Quantity                         26026133   
Fat supply quantity (g/capita/day)       3435732   
Feed                                     7282199   
Food                                    25406622   
Food supply (kcal/capita/day)            3329296   
Food supply quantity (kg/capita/yr)      3163725   
Import Quantity                         28834929   

                                                                                  Unit  \
                                                                                   sum   
Element                                                                                  
Domestic supply quantity             1000 tonnes1000 tonnes1000 tonnes1000 tonnes10...   
Export Quantity                      1000

# Answer. highest sum of import quantity is year 2017 with sum = 294559.09

# Question 16. Total number of the sum of production in 2014.

In [61]:
Sum_of_Production = data.groupby('Element').agg(['sum', "count"])

In [66]:
Sum_of_Production.head(12)

Area Code        \
                                          sum count   
Element                                               
Domestic supply quantity               708993  5295   
Export Quantity                        599910  4403   
Fat supply quantity (g/capita/day)     675050  5023   
Feed                                   176272  1319   
Food                                   663295  4941   
Food supply (kcal/capita/day)          674057  5014   
Food supply quantity (kg/capita/yr)    658446  4905   
Import Quantity                        688174  5139   
Losses                                 274353  2009   
Other uses (non-food)                  235554  1732   
Processing                             271940  2010   
Production                             526751  3881   

                                                                                  Area  \
                                                                                   sum   
Element                                                                                  
Domestic supply quantity             AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Export Quantity                      AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Fat supply quantity (g/capita/day)   AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Feed                                 AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Food                                 AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Food supply (kcal/capita/day)        AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Food supply quantity (kg/capita/yr)  AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Import Quantity                      AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Losses                               AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Other uses (non-food)                AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Processing                           AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   
Production                           AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...   

                                          Item Code        \
                                    count       sum count   
Element                                                     
Domestic supply quantity             5295  14197445  5295   
Export Quantity                      4403  11840553  4403   
Fat supply quantity (g/capita/day)   5023  13535000  5023   
Feed                                 1319   3538507  1319   
Food                                 4941  13285035  4941   
Food supply (kcal/capita/day)        5014  13511060  5014   
Food supply quantity (kg/capita/yr)  4905  13185401  4905   
Import Quantity                      5139  13795966  5139   
Losses                               2009   5424803  2009   
Other uses (non-food)                1732   4729749  1732   
Processing                           2010   5350416  2010   
Production                           3881  10450053  3881   

                                                                                  Item  \
                                                                                   sum   
Element                                                                                  
Domestic supply quantity             PopulationCereals - Excluding BeerWheat and pr...   
Export Quantity                      Cereals - Excluding BeerWheat and productsRice...   
Fat supply quantity (g/capita/day)   Grand TotalVegetal ProductsAnimal ProductsCere...   
Feed                                 Cereals - Excluding BeerWheat and productsRice...   
Food                                 Cereals - Excluding BeerWheat and productsRice...   
Food supply (kcal/capita/day)        Grand TotalVegetal ProductsAnimal ProductsCere...   
Food supply quantity (kg/capita/yr)  Cereals - Excluding BeerWheat and productsRice...   
Import Quantity                      Cereal

# Total Production for 2014 = 1931287.75

# Question 17. Element with highest sum in 2018

In [68]:
Element_with_highest_sum = data.groupby('Element').agg(['sum'])

In [78]:
Element_with_highest_sum.Y2018

sum
Element                                           
Domestic supply quantity                2161192.10
Export Quantity                          181594.80
Fat supply quantity (g/capita/day)        10258.69
Feed                                     233489.68
Food                                    1303841.28
Food supply (kcal/capita/day)            455261.00
Food supply quantity (kg/capita/yr)       49056.85
Import Quantity                          287997.09
Losses                                   163902.00
Other uses (non-food)                     91300.97
Processing                               308429.00
Production                              2075072.89
Protein supply quantity (g/capita/day)    11833.56
Residuals                                 34864.00
Seed                                      25263.14
Stock Variation                           20577.91
Total Population - Both sexes           1140605.00
Tourist consumption                          90.00

# Answer = Domestic supply quantity	2161192.10

# Question 18. Element with 3rd lowest sum in 2018

In [79]:
Element_with_highest_sum = data.groupby('Element').agg(['sum'])

In [83]:
Element_with_highest_sum.Y2018

sum
Element                                           
Domestic supply quantity                2161192.10
Export Quantity                          181594.80
Fat supply quantity (g/capita/day)        10258.69
Feed                                     233489.68
Food                                    1303841.28
Food supply (kcal/capita/day)            455261.00
Food supply quantity (kg/capita/yr)       49056.85
Import Quantity                          287997.09
Losses                                   163902.00
Other uses (non-food)                     91300.97
Processing                               308429.00
Production                              2075072.89
Protein supply quantity (g/capita/day)    11833.56
Residuals                                 34864.00
Seed                                      25263.14
Stock Variation                           20577.91
Total Population - Both sexes           1140605.00
Tourist consumption                          90.00

# Answer. SEED is the 3rd lowest sum in 2018 with 25263.14